src/main/java/com/hrznstudio/galacticraft/block/special/walkway/WireWalkway.java

```diff
@@ -0,0 +1,289 @@
+                i |= getFacingMask(Direction.SOUTH);
+            if (blockState.get(FACING).equals(Direction.EAST))
+                i |= getFacingMask(Direction.EAST);
+            if (blockState.get(FACING).equals(Direction.WEST))
+                i |= getFacingMask(Direction.WEST);
+            if (blockState.get(FACING).equals(Direction.UP))
+                i |= getFacingMask(Direction.UP);
+            if (blockState.get(FACING).equals(Direction.DOWN))
+                i |= getFacingMask(Direction.DOWN);
+            return i;
+        });
+    }
+
+    private BooleanProperty getPropForDir(Direction direction) {
+        switch (direction) {
+            case NORTH:
+                return NORTH;
+            case WEST:
+                return WEST;
+            case EAST:
+                return EAST;
+            case SOUTH:
+                return SOUTH;
+            case UP:
+                return UP;
+            case DOWN:
+                return DOWN;
+        }
+        throw new IllegalArgumentException();
+    }
+
+    @Override
+    public BlockState rotate(BlockState state, BlockRotation rotation) {
+        switch (rotation) {
+            case CLOCKWISE_180:
+                return state.with(NORTH, state.get(SOUTH)).with(EAST, state.get(WEST)).with(SOUTH, state.get(NORTH)).with(WEST, state.get(EAST));
+            case COUNTERCLOCKWISE_90:
+                return state.with(NORTH, state.get(EAST)).with(EAST, state.get(SOUTH)).with(SOUTH, state.get(WEST)).with(WEST, state.get(NORTH));
+            case CLOCKWISE_90:
+                return state.with(NORTH, state.get(WEST)).with(EAST, state.get(NORTH)).with(SOUTH, state.get(EAST)).with(WEST, state.get(SOUTH));
+            default:
+                return state;
+        }
+    }
+
+    @Override
+    public BlockState mirror(BlockState state, BlockMirror mirror) {
+        switch (mirror) {
+            case LEFT_RIGHT:
+                return state.with(NORTH, state.get(SOUTH)).with(SOUTH, state.get(NORTH));
+            case FRONT_BACK:
+                return state.with(EAST, state.get(WEST)).with(WEST, state.get(EAST));
+            default:
+                return super.mirror(state, mirror);
+        }
+    }
+
+    public boolean canConnect(BlockState state, BlockState neighborState, BlockPos pos, BlockPos neighborPos, WorldAccess world) {
+        try {
+            if (pos.offset(state.get(FACING)).equals(neighborPos))
+                return false;
+            if (neighborPos.offset(neighborState.get(FACING)).equals(pos))
+                return false;
+        } catch (IllegalArgumentException ignored) {}
+        // TODO: Only connect to machines when the wire port is facing the block and the block isn't facing the machine
+        // TODO: The WireBlockEntity will still connect on the top face of this block (there's no wire there)
+        return neighborState.getBlock() instanceof WireBlock || neighborState.getBlock() instanceof ConfigurableMachineBlock;
```

Comment:
Use `ComponentHelper.CAPACITOR.hasComponent` - otherwise they'll only connect to our devices.